.github/workflows/ci.yaml

```diff
@@ -148,7 +150,17 @@ jobs:
             findRes=$(find "${pkgFolder}" -name "*.cy.ts" || true)
             if [[ -n "${findRes}" ]]; then
                 componentTestFilter="${componentTestFilter}${pkgFolder},"
-                componentTestMatrix="${componentTestMatrix}\"${pkgFolder}\","
+
+                # if there is cypress.config in the package, that means it should run tests independently.
+                # we use `pnpm run test:component:ci --project <pkgFolder>` instead.
+                # so we will not add it to the componentTestMatrix, rather we will add it to a matrix.include
+                # that we can detect it later
+                findCyConfig=$(find "${pkgFolder}" -name "cypress.config.ts" || true)
+                if [[ -z "${findCyConfig}" ]]; then
+                    componentTestMatrix="${componentTestMatrix}\"${pkgFolder}\","
```

Comment:
We should always fill the `componentTestMatrix` and add conditional flag into the `componentTestMatrixInclude`